sgt-back-end/server.js

```diff
@@ -0,0 +1,162 @@
+import express from 'express';
+import pg from 'pg';
+
+const app = express();
+app.use(express.json());
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades/:gradeId', async (req, res) => {
+  try {
+    const gradeId = Number(req.params.gradeId);
+    if (!Number.isInteger(gradeId) || gradeId <= 0) {
+      res.status(400).json({ error: '"gradeId" must be a positive integer' });
+      return;
+    }
+    const sql = `
+      select *
+        from "grades"
+       where "gradeId" = $1
+    `;
+    const params = [gradeId];
+    const result = await db.query(sql, params);
+    const grade = result.rows[0];
+    if (grade) {
+      res.json(grade);
+    } else {
+      res.status(404).json({ error: `Cannot find grade with "gradeId" ${gradeId}` });
+    }
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ error: 'An unexpected error occurred.' });
+  }
+});
+
+app.get('/api/grades', async (req, res) => {
+  try {
+    const sql = `
+      select *
+        from "grades";
+    `;
+    const result = await db.query(sql);
+    const gradesArray = result.rows;
+    res.json(gradesArray);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ error: 'An unexpected error occurred.' });
+  }
+});
+
+app.post('/api/grades', async (req, res) => {
+  try {
+    const gradeInput = req.body;
```

Comment:
It would make the code a bit nicer if you were to destructure `req.body` into the variables you are using. Then you wouldn't need to repeat `gradeInput.` so frequently.

Though I see that if you do that you won't be able to use your fancy loop to check missing input. You could do the loop and then destructure after the loop.